graphql/schema/query.graphql

```diff
@@ -94,6 +89,6 @@ type Query {
   taskNamesForBuildVariant(projectIdentifier: String!, buildVariant: String!): [String!]
 
   # version
-  hasVersion(id: String!): Boolean!
-  version(id: String!): Version!
+  hasVersion(id: String @deprecated(reason: "use patchId instead"), patchId: String): Boolean!
+  version(id: String @deprecated(reason: "use versionId instead"), versionId: String): Version!
```

Comment:
For clarification, will this change ever be safe to make in a backwards compatible way? It seems like at some point we'll need to deploy a version of the backend that only has one argument and would need to update the front end at the same time.

Reply:
oh, I was going to deploy this, then change the frontend to only use the new args (e.g. `patchId`). Do you think that wouldn't work? (I also edited the resolvers) 👀

Reply:
Would it be possible for the front end to only use the new arg? I thought since the query args are order dependent, we'd still need to have `null` or an empty string as the legacy `id` argument, which would make it impossible to remove from the backend safely. But I may be wrong 😳

Reply:
oh!! that's a good point! I tested it out and it seems ok but let me know if it seems wack 🫣

<img width="1065" alt="Screenshot 2024-03-26 at 3 02 52 PM" src="https://github.com/evergreen-ci/evergreen/assets/47064971/662f4c02-3997-4ff3-bd63-b19d1648f3ea">

<img width="1069" alt="Screenshot 2024-03-26 at 3 08 20 PM" src="https://github.com/evergreen-ci/evergreen/assets/47064971/d69e34bb-66d2-4132-be1e-e6c512e6be55">

Reply:
oh that's awesome, you learn something new every day! thanks for looking into it.